File: basic_rospy/src/simple_node_pub_obj.py
#! /usr/bin/env python3

import rospy
from std_msgs.msg import Int32

class Publisher():
    def __init__(self):
        self.pub = rospy.Publisher("/contador", Int32, queue_size=1)
        self.msg = Int32()

    def doWork(self):
        self.pub.publish(self.msg)
        self.msg.data += 1

def main():
    rospy.init_node("simple_node_pub_obj")

    pub = Publisher()

    rate = rospy.Rate(2)
    while(not rospy.is_shutdown()):
        pub.doWork()
        rate.sleep()

if __name__ == "__main__":

    try:
        main()
    except rospy.ROSInterruptException:
        pass

File: basic_roscpp/src/simple_node_pub_obj.cpp
// Copyright 2020 Juan Carlos Manzanares Serrano

#include "ros/ros.h"
#include "std_msgs/Int64.h"

class Publicador
{
public:
    Publicador()
    {
        pub = n.advertise<std_msgs::Int64>("/contador", 1);
    }

    void do_work()
    {
        std_msgs::Int64 msg;

        msg.data = count++;

        pub.publish(msg);
    }

private:
    ros::NodeHandle n;

    int count = 0;

    ros::Publisher pub;
};

int main(int argc, char **argv)
{
    ros::init(argc, argv, "simple_node_pub_obj");

    Publicador pub;

    ros::Rate loop_rate(20);

    while (ros::ok())
    {
        pub.do_work();

        ros::spinOnce();
        loop_rate.sleep();
    }

    return 0;
}